common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -105,7 +105,7 @@ public void fetchBlocks(
           (blockIds1, listener1) -> {
             // Unless this client is closed.
             if (clientFactory != null) {
-              TransportClient client = clientFactory.createClient(host, port);
+              TransportClient client = clientFactory.createClient(host, port, true);
```

Comment:
this is being passed in all the time even when the maxRetries is 0.  change the parameter to be based on:
if (maxRetries > 0) true else false